File: app/src/main/java/ithaca_transit/android/cornellappdev/com/ithaca_transit/Models/BusPath.java
// Path.java
package ithaca_transit.android.cornellappdev.com.ithaca_transit.Models;

import android.graphics.Color;
import com.google.android.gms.maps.model.PolylineOptions;
import kotlin.jvm.internal.Intrinsics;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;


public final class BusPath extends Path {
    @NotNull
    private Integer[] dashLengths;
    private Integer[] dashColors;
    private Float polylineWidth;
    private List traveledPath;
    private List untraveledPath;


    public BusPath(@NotNull Waypoint[] waypoints) {
        super(waypoints);
        this.dashLengths = new Integer[]{this.getDASH_LENGTHS_X(), this.getDASH_LENGTHS_Y()};
        this.dashColors = new Integer[]{this.getColor(), this.getBLANK_COLOR()};
        this.polylineWidth = this.getLINE_WIDTH();
        this.createPath(waypoints);

        this.untraveledPath = super.getTraveledPolyline().getPoints();
        this.traveledPath = this.untraveledPath;

        super.getTraveledPolyline().color(super.getColor());
        super.getTraveledPolyline().width(polylineWidth);
    }


    public final void createPath(@NotNull Waypoint[] waypoints) {
        Intrinsics.checkParameterIsNotNull(waypoints, "waypoints");

        for(int k = 0; k < waypoints.length; ++k) {
            Waypoint point = waypoints[k];
            getTraveledPolyline().add(point.coordinates());
        }
    }
}

File: app/src/main/java/ithaca_transit/android/cornellappdev/com/ithaca_transit/Models/BusPath.kt
package ithaca_transit.android.cornellappdev.com.ithaca_transit.Models

import android.graphics.Color
import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.PolylineOptions

enum class PathType {
    driving, walking
}

open class Path(waypoints: Array<Waypoint>) {
    val BLANK_COLOR = Color.rgb(0, 0, 0)
    val DASH_LENGTHS_X = 6
    val DASH_LENGTHS_Y = 4
    val LINE_WIDTH = 8.0F

    var waypoints: Array<Waypoint>? = null
    var traveledPolyline: PolylineOptions? = null
    var color: Int

    init {
        this.waypoints = waypoints
        this.traveledPolyline = PolylineOptions()
        this.color = BLANK_COLOR
    }
}

class BusPath(waypoints: Array<Waypoint>) : Path(waypoints) {

    var dashLengths: Array<Int> = arrayOf(DASH_LENGTHS_X, DASH_LENGTHS_Y)
    private var dashColors: Array<Int> = emptyArray()
    private var polylineWidth: Float?
    private var traveledPath: List<LatLng>? = null
    private var untraveledPath: List<LatLng>? = null

    init {
//        super.color = ContextCompat.getColor(R.colors.blue);
        this.dashColors = arrayOf(color!!, BLANK_COLOR)
        this.polylineWidth = LINE_WIDTH
        createPath(waypoints)
        this.untraveledPath = super.traveledPolyline!!.points
        this.traveledPath = untraveledPath
        super.traveledPolyline!!.color(super.color)
        super.traveledPolyline!!.width(polylineWidth!!)
    }

    fun createPath(waypoints: Array<Waypoint>) {
        var count = 0
        for (point in waypoints) {
            this.traveledPolyline!!.add(point.coordinates())
            count = count + 1
        }
    }
    
}